libs/community/langchain_community/vectorstores/vdms.py

```diff
@@ -0,0 +1,1630 @@
+from __future__ import annotations
+
+import base64
+import logging
+import uuid
+from copy import deepcopy
+from enum import Enum
+from typing import (
+    Any,
+    Callable,
+    Dict,
+    Iterable,
+    List,
+    Optional,
+    Sized,
+    Tuple,
+    Type,
+    Union,
+)
+
+import numpy as np
+from langchain_core.documents import Document
+from langchain_core.embeddings import Embeddings
+from langchain_core.vectorstores import VectorStore
+
+from langchain_community.vectorstores.utils import maximal_marginal_relevance
+
+try:
+    import vdms
+except ImportError:
+    raise ImportError(
+        "Could not import vdms python package. "
+        "Please install it with `pip install vdms."
+    )
+
+
+class DistanceMetric(str, Enum):
```

Comment:
It's generally recommended to rely on Literal instead of Enum for things that are user facing -- it makes it easier to use the code without having to fish for additional imports. Enums are still good for internal purposes.

I'd recommend updating the init to use `Literal['L2', 'IP']` instead of DistanceMetric

Reply:
Changed to using Literal instead of DistanceMetric